src/agent/src/agents/mod.rs

```diff
@@ -13,8 +18,14 @@ pub struct AgentOutput {
 }
 
 pub trait Agent {
-    fn prepare(&self) -> AgentResult<AgentOutput>;
-    fn run(&self) -> AgentResult<AgentOutput>;
+    fn prepare<'a>(
+        &'a self,
+        child_processes: &'a Arc<Mutex<HashSet<u32>>>,
+    ) -> Pin<Box<dyn Future<Output = AgentResult<AgentOutput>> + Send + '_>>;
+    fn run<'a>(
+        &'a self,
+        child_processes: &'a Arc<Mutex<HashSet<u32>>>,
+    ) -> Pin<Box<dyn Future<Output = AgentResult<AgentOutput>> + Send + '_>>;
```

Comment:
is there a reason to add futures?

Reply:
The functions needs to lock a Mutex, so they need to be async. 
The functions now returns a future in a pinned box to ensure object safety.